### PR TITLE
Restructure Dataset

### PR DESCRIPTION
**Pull request by philippmwirth**
# Restructure Dataset and Collate Function

To make it easier to understand the code and the package, we have decided to change the collate functions in such a way that they return a tuple of transformed images instead of a single batch with twice the batch size, i.e. the new collate function returns:
```python
batch = collate_fn(...)
(transformed_0, transformed_1), labels, filenames = batch
```
All of the models and loss functions were also adjusted so that the new workflow reads:
```python
# schematic workflow for training

for e in range(1, epochs+1):
   
    for (x0, x1), _, _ in dataloader:
        ...
        
        # forward pass of the transformed images
        y0, y1 = model(x0, x1)

        # calculate loss from the projections of the transformed images
        loss = loss_fn(y0, y1)

        ...

```

Furthermore, I picked up some issues along the way, including: 
- #84 Error while using VideoDataset without extension specified 
- Remove dead code (e.g. the VAEEmbedding)
- Clean up autocomplete

## TODO
- [x] Update docstrings
- [x] Benchmark the models to make sure they still work
- [x] Update the documentation

### CODE REVIEW

**Review by IgorSusmelj — APPROVED** [review]

**Comment by IgorSusmelj** [comment]
Don't forget to update the code for the tutorials. Or do you want me to have a look at them when the PR is ready?

**Comment by philippmwirth** [comment]
> 
> 
> Don't forget to update the code for the tutorials. Or do you want me to have a look at them when the PR is ready?

I will change the code for the tutorials and then you can take a look at them in the PR.

File: docs/source/tutorials_source/package/tutorial_simclr_clothing.py [hunk 1]
```diff
@@ -167,14 +167,14 @@
...
-                                  max_epochs=max_epochs)
+encoder.train_embedding(gpus=gpus, 
+                        progress_bar_refresh_rate=100,
+                        max_epochs=max_epochs)
 
 # %%
 # Now, let's make sure we move the trained model to the gpu if we have one
 device = 'cuda' if gpus==1 else 'cpu'
-encoder.to(device)
+encoder = encoder.to(device)
```
Comment (IgorSusmelj) [inline docs/source/tutorials_source/package/tutorial_simclr_clothing.py:177]: I think this is not needed

Comment (philippmwirth) [reply docs/source/tutorials_source/package/tutorial_simclr_clothing.py:177]: yep

**Review by IgorSusmelj — APPROVED** [review]
